app/Http/Controllers/Dashboard.php

```diff
@@ -19,7 +23,120 @@ public function index(): void
       Redirect::to('/login');
     }
 
-    
-    $this->view('dashboard/index');
+    $data = $this->getSidebarData();
+    $data['table_contents'] = $this->getTableContent();
+    $data['encoding_types'] = $this->getEncodingTypes();
+
+    $this->view('dashboard/main', $data);
+  }
+
+  /**
+   * Fetches all data related to the sidebar
+   *
+   * @return array
+   */
+  protected function getSidebarData(): array
+  {
+    $db = Database::getInstance();
+
+    try {
+      $tables = $db->getTables();
+    } catch (\Throwable $th) {
+      session('db_err', $th->getMessage());

```

Comment:
It's a good practice to avoid using abbreviations and use the full words, so in this case `db_error` would be more appropriate